src/services/pathingService.ts

```diff
@@ -32,13 +36,64 @@ export default class PathingService implements IPathingService {
         this.setUnwalkable(emptyGrid, innerPolygon);
 
         // Calculate the path to take
-        let bloc = boothLocatiob ? boothLocatiob : { x: 0, y: 0 };
-        let rawPath = finder.findPath(bloc.x, bloc.y, targetLocation.x, targetLocation.y, emptyGrid);
+        let bloc = boothLocation ? boothLocation : { x: 0, y: 0 };
+
+        let shortestLength = this.calculateLength(bloc, targetLocation);
+        let alternative = await this.selectPath(bloc, targetLocation);
+
+        let smoothedPath: any[] = [];
 
-        // Smooth out the path
-        return pathfinder.Util.smoothenPath(emptyGrid, rawPath);
-        // Render & animate the path
+        if (alternative && shortestLength * alternative.path.weight > alternative.length) {
+            let targetX = parseInt(alternative.path.location.x);
+            let targetY = parseInt(alternative.path.location.y);
 
+            finder.findPath(bloc.x, bloc.y, targetX, targetY, emptyGrid);
+            let rawPath = finder.findPath(targetX, targetY, targetLocation.x, targetLocation.y, emptyGrid);
+
+            smoothedPath.push(pathfinder.Util.smoothenPath(emptyGrid, rawPath));
+            return smoothedPath;
+        } else {
+            let rawPath = finder.findPath(bloc.x, bloc.y, targetLocation.x, targetLocation.y, emptyGrid);
+            smoothedPath.push(pathfinder.Util.smoothenPath(emptyGrid, rawPath));
+            return smoothedPath;
+        }
+    }
+
+    /**
+     * This function is responsible for determing wether or not the magnetic path has priority or the standard path. 
+     * @param start The Start Coordinates
+     * @param end The End Coordinates
+     */
+    private selectPath = async (start: ICoord, end: ICoord) => {
+        let magnetizedItems = await this.magnetService.getMagneticProducts();
+        let filteredItems = magnetizedItems.filter(item => {
+            if (item.weight > 1) {
```

Comment:
What if two routes have the same length, but one of them have a magnetized product?